File: ios/Samples/HelloModule.m
//
//  RCTHelloModule.h
//  NativeModules
//
//  Created by Uday Pandey on 18/09/2023.
//

#ifndef RCTHelloModule_h
#define RCTHelloModule_h


// CalendarManagerBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(HelloModule, NSObject)

RCT_EXTERN_METHOD(hello:(NSString *)name)

@end

#endif /* RCTHelloModule_h */

File: ios/Samples/HelloModule.swift
//
//  RCTHelloModule.swift
//  NativeModules
//
//  Created by Uday Pandey on 18/09/2023.
//

import Foundation

// HelloModule.swift

@objc(HelloModule)
class HelloModule: NSObject {
  
  @objc(hello:)
  func hello(_ name: String) -> Void {
    print("Got name: \(name)")
  }
  
  @objc
  class func requiresMainQueueSetup() -> Bool {
    false
  }
}
